src/shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberColumnFormatter.tsx

```diff
@@ -0,0 +1,192 @@
+import { ASCNCopyNumberValue } from 'shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberElement';
+import { ASCN_BLACK } from 'shared/lib/Colors';
+import { getASCNCopyNumberColor } from 'shared/lib/ASCNUtils';
+
+/**
+ * @author Avery Wang
+ */
+
+// gets value displayed in table cell - "NA" if missing attributes needed for calculation
+function getAscnCopyNumberData(
+    mutation: Mutation,
+    sampleIdToClinicalDataMap:
+        | { [sampleId: string]: ClinicalData[] }
+        | undefined
+) {
+    return hasASCNProperty(mutation, 'ascnIntegerCopyNumber')
+        ? mutation.alleleSpecificCopyNumber.ascnIntegerCopyNumber
+        : ASCNCopyNumberValue.NA;
+}
+
+// sort by total copy number (since that is the number displayed in the icon
+function getAllTotalCopyNumberForMutation(
+    data: Mutation[],
+    sampleIdToClinicalDataMap: { [key: string]: ClinicalData[] } | undefined,
+    sampleIds: string[]
+) {
+    const sampleToCNA: { [key: string]: string } = _
+        .chain(data)
+        .keyBy('sampleId')
+        .mapValues(function(mutation) 
+            {
+                let ascnCopyNumberValue = getAscnCopyNumberData(mutation, sampleIdToClinicalDataMap);
+                if (
+                    ascnCopyNumberValue !== ASCNCopyNumberValue.NA &&
+                    hasASCNProperty(mutation, 'totalCopyNumber') &&
+                    getWGD(sampleIdToClinicalDataMap, mutation.sampleId) !== ASCNCopyNumberValue.NA &&
+                    getASCNCopyNumberColor(ascnCopyNumberValue.toString()) !== ASCN_BLACK
+                ) {
+                    return mutation.alleleSpecificCopyNumber.totalCopyNumber.toString();
+                }
+                return ASCNCopyNumberValue.NA;
+            })
+        .value();
+    return sampleToCNA;
+}
+
+function getSortValue(
+    data: Mutation[],
+    sampleIdToClinicalDataMap: { [key: string]: ClinicalData[] } | undefined,
+    sampleIds: string[]
+) {
+    const displayValuesBySample: {
+        [key: string]: string;
+    } = getAllTotalCopyNumberForMutation(
+        data,
+        sampleIdToClinicalDataMap,
+        sampleIds
+    );
+    const sampleIdsWithValues = sampleIds.filter(
+        sampleId => displayValuesBySample[sampleId]
+    );
+    const displayValuesAsString = sampleIdsWithValues.map(
+        (sampleId: string) => {
+            return displayValuesBySample[sampleId];
+        }
+    );
+    return displayValuesAsString.join(';');
+}
+
+export function getWGD(
+    sampleIdToClinicalDataMap:
+        | { [sampleId: string]: ClinicalData[] }
+        | undefined,
+    sampleId: string
+) {
+    let wgdData = sampleIdToClinicalDataMap
+        ? sampleIdToClinicalDataMap[sampleId].filter(
+              (cd: ClinicalData) => cd.clinicalAttributeId === 'FACETS_WGD'
+          )
+        : undefined;
+    return wgdData !== undefined && wgdData.length > 0
+        ? wgdData[0].value
```

Comment:
Are we expecting to get multiple wgdData? If not, you could consider a keyBy on `sampleIdToClinicalDataMap[sampleId]` then take the value under `FACETS_WGD `

Reply:
I can't recall.. @averyniceday @n1zea144 do you recall?

Reply:
@averyniceday Is there a reason why we should not check if there are multiple values? We are trying to decide on whether this change needs to be made or not.